server/src/main/java/org/eclipse/hono/server/MessageForwardingEndpoint.java

```diff
@@ -38,13 +40,24 @@
  */
 public abstract class MessageForwardingEndpoint extends BaseEndpoint {
 
+    protected CounterService              counterService;
```

Comment:
I think you are right that for this trivial interfaces the argument against NullObjects are not valid. And sure - a if() check on null does not look very nice. How would you create and assign such NullObjects in our environment? - maybe I oversee a good mechanism, here.